src/Microsoft.AspNetCore.Mvc.TagHelpers/Cache/IHtmlFragmentCache.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Html;
+
+namespace Microsoft.AspNetCore.Mvc.TagHelpers.Cache
+{
+    public interface IHtmlFragmentCache
+    {
+        Task<IHtmlContent> SetAsync(object key, Func<Task<IHtmlContent>> renderContent, HtmlFragmentCacheContext context);
```

Comment:
Could be right. Also if no key is provided the tag helper generates one automatically, which can't be distributed, but is also a string. Will change.